From 175c68f2423b58d0f9d670cfb148eea2bddf2043 Mon Sep 17 00:00:00 2001
From: "Guoniu.zhou" <guoniu.zhou@nxp.com>
Date: Wed, 8 May 2024 11:11:33 +0800
Subject: [PATCH] LF-10512: media: imx: ci_pi: fix green line at frame bottom
 issue
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The bottom line of image will be green since the data are zero.
The reason is due to vsync signal polarity mismatch between camera
and CI_PI expected.

The CI_PI expected vsync signal like this:
        ┌─┐                      ┌─┐
	│ │                      │ │
VSYNC ──┘ └──────────────────────┘ └──

The vsync signal of camera used on iMX93/91 like this:

VSYNC ──┐ ┌──────────────────────┐ ┌──
        │ │                      │ │
	└─┘                      └─┘

So need to invert the vsync signal polarity in CI_PI side.

Upstream-Status: Backport [6.6-2.0.x-imx]

Signed-off-by: Guoniu.zhou <guoniu.zhou@nxp.com>
Reviewed-by: Sandor Yu <Sandor.yu@nxp.com>
Acked-by: Jason Liu <jason.hui.liu@nxp.com>
---
 drivers/staging/media/imx/imx8-parallel-csi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/media/imx/imx8-parallel-csi.c b/drivers/staging/media/imx/imx8-parallel-csi.c
index bf6d90b93faa..06e653a9797a 100644
--- a/drivers/staging/media/imx/imx8-parallel-csi.c
+++ b/drivers/staging/media/imx/imx8-parallel-csi.c
@@ -990,7 +990,7 @@ static const struct mxc_pcsi_plat_data imx93_pdata = {
 	.interface_ctrl_reg1	= 0xc,
 	.has_gasket		= true,
 	.def_hsync_pol		= 0,
-	.def_vsync_pol		= 0,
+	.def_vsync_pol		= 1,
 	.def_pixel_clk_pol	= 0,
 	.def_csi_in_data_type	= CSI_IN_DT_RAW_10,
 	.pd_ops			= NULL,
-- 
2.39.3

